src/utils/flume/mod.rs

```diff
@@ -12,3 +34,280 @@ impl<T> FlumeSenderExt<T> for flume::Sender<T> {
         }
     }
 }
+
+pub fn unbounded_with_default_metrics_registry<T>(
+    name: Cow<'static, str>,
+) -> (SizeTrackingSender<T>, SizeTrackingReceiver<T>) {
+    unbounded_with_metrics_registry(name, &mut default_registry())
+}
+
+pub fn unbounded_with_metrics_registry<T>(
+    name: Cow<'static, str>,
+    registry: &mut Registry,
+) -> (SizeTrackingSender<T>, SizeTrackingReceiver<T>) {
+    let (sender, receiver) = flume::unbounded();
+    new_with_metrics_registry(name, registry, sender, receiver, None)
+}
+
+pub fn bounded_with_default_metrics_registry<T>(
+    capacity: usize,
+    name: Cow<'static, str>,
+) -> (SizeTrackingSender<T>, SizeTrackingReceiver<T>) {
+    bounded_with_metrics_registry(capacity, name, &mut default_registry())
+}
+
+pub fn bounded_with_metrics_registry<T>(
+    capacity: usize,
+    name: Cow<'static, str>,
+    registry: &mut Registry,
+) -> (SizeTrackingSender<T>, SizeTrackingReceiver<T>) {
+    let (sender, receiver) = flume::bounded(capacity);
+    new_with_metrics_registry(name, registry, sender, receiver, Some(capacity))
+}
+
+fn new_with_metrics_registry<T>(
+    name: Cow<'static, str>,
+    registry: &mut Registry,
+    sender: flume::Sender<T>,
+    receiver: flume::Receiver<T>,
+    capacity: Option<usize>,
+) -> (SizeTrackingSender<T>, SizeTrackingReceiver<T>) {
+    static ID_GENERATOR: AtomicUsize = AtomicUsize::new(0);
+
+    let id = ID_GENERATOR.fetch_add(1, Ordering::Relaxed);
+    let tracker = ChannelMemoryUsageTracker {
+        id,
+        name,
+        capacity,
+        ..Default::default()
+    };
+    registry.register_collector(Box::new(tracker.clone()));
+
+    (
+        SizeTrackingSender {
+            sender,
+            tracker: tracker.clone(),
+        },
+        SizeTrackingReceiver { receiver, tracker },
+    )
+}
+
+#[derive(Debug)]
```

Comment:
It'd be great to put some docs around this (even with doc examples), especially given its ubiquitous usage across the codebase. Same goes for other introduces `SizeTracking*` data structures.

I'd also consider putting them (flume, cache and I think there was one more) in a dedicated module, i.e., `src/utils/size_tracking/*`. This could be done in a separate PR if you decide it makes sense.